python/lsst/ts/Dome/mock_controller.py

```diff
@@ -207,13 +204,22 @@ async def cmd_loop(self, reader, writer):
                     else:
                         func = self.dispatch_dict[cmd]
                         kwargs = items["parameters"]
-                        if cmd[:6] == "status":
+                        if cmd.startswith("status"):
                             # the status commands take care of sending a reply
                             # themselves
                             send_response = False
-                        if cmd == "stopAz" or cmd == "stopEl":
-                            timeout = self.short_timeout
-                        await func(**kwargs)
+                        if cmd in [
+                            "moveAz",
+                            "crawlAz",
+                            "stopAz",
+                            "moveEl",
+                            "crawlEl",
+                            "stopEl",
+                            "park",
+                        ]:
+                            timeout = await func(**kwargs)
+                        else:
+                            await func(**kwargs)
```

Comment:
Thanks for that great suggestion. I had to do a slightly different implementation because ALL commands should return a duration but not all of them compute one at the moment. So now, if duration == None the code sets a default duration instead.

Regarding renaming timeout to duration: you are absolutely right. Unfortunately I cannot rename the parameter "timeout" without discussing with EIE first so I have only renamed the variables for now and added a note to rename the parameter as well as soon as EIE agrees.